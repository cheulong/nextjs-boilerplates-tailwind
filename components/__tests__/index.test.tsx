import { render, screen, fireEvent } from '@testing-library/react';
import Home from '@/pages/index';
import I18nProvider from 'next-translate/I18nProvider';
import commonEN from '../../locales/en/common.json';

describe('Home', () => {
  it('renders a home page', () => {
    const { container } = render(
      <I18nProvider lang="en" namespaces={{ common: commonEN }}>
        <Home />
      </I18nProvider>,
    );

    const heading = screen.getByText('Hello world!');
    expect(heading).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
