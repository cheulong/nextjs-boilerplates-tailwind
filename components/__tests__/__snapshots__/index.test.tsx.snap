// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a home page 1`] = `
<div>
  <div>
    <h1
      class="text-3xl font-bold underline"
    >
      Navbar!
    </h1>
    <button>
      Japanese
    </button>
    <br />
    <button>
      English
    </button>
    <br />
    <button>
      Korean
    </button>
    <h1
      class="title"
    >
      Hello world!
    </h1>
    <button
      class="wrapper"
    >
      BaseButton
    </button>
  </div>
</div>
`;
